### ISSUE TITLE
dht2 fails when handed a ddf object with uniform key function

### ISSUE THREAD

**Issue by erex** [DistanceDevelopment/Distance#88]
Email to [list](https://groups.google.com/g/distance-sampling/c/Jm3M1PajDtw) 31May21 showed error thrown when dht2 was handed a ddf object with uniform key.

```
Error in `[.data.frame`(newdata, , c("distance",fvars), drop = FALSE) : object 'fvars' not found
```

Error is actually thrown by `predict.ds` on [line 176](https://github.com/DistanceDevelopment/mrds/blob/f0510527f56a9ec92df8c9665f40a5b9f4e06840/R/predict.ds.R#L176) likely becausee the loop starting on [line 163](https://github.com/DistanceDevelopment/mrds/blob/f0510527f56a9ec92df8c9665f40a5b9f4e06840/R/predict.ds.R#L163) is not executed when there is no `shape` or `scale` parameter in the key function (such as the uniform).

**Comment by dill** [comment]
Trying to reproduce this with `minke`...

```r
library(Distance)
data(minke)

minke_df <- ds(minke, truncation=1.5, adjustment="cos", key="unif", order=1)

minke_dht2 <- dht2(minke_df, flatfile=minke, stratification="geographical", strat_formula=~1)
```

I don't think that your diagnosis above is correct as at [165](https://github.com/DistanceDevelopment/mrds/blob/f0510527f56a9ec92df8c9665f40a5b9f4e06840/R/predict.ds.R#L165) the inside of the loop will be skipped if there are no shape or scale pars in the model and things proceed straight to [206](https://github.com/DistanceDevelopment/mrds/blob/f0510527f56a9ec92df8c9665f40a5b9f4e06840/R/predict.ds.R#L206).

Is there something else unusual about this dataset/model?

**Comment by dill** [comment]
Oh, sorry I see this next code block regarding binned data, I'll continue to investigate...

**Comment by dill** [comment]
fixed in `mrds` [327751a](https://github.com/DistanceDevelopment/mrds/commit/327751a6bf02c23c3d62ace9327ef018af3fdb45)
